build fails with WITH_CHECK=OFF

I test SuiteSparse-5.13.0-cmake.3 on macOS.

In build `cmake .. -DWITH_DEMOS=OFF` works, however
`cmake .. -DWITH_DEMOS=OFF -DWITH_CHECK=OFF`
results in
```
Consolidate compiler generated dependencies of target umfpack_int
[ 26%] Building C object CMakeFiles/umfpack_int.dir/UMFPACK/Source/umf_cholmod.c.o
/Users/fwein/Downloads/SuiteSparse-5.13.0-cmake.3/UMFPACK/Source/umf_cholmod.c:224:9: error: implicit declaration of function 'cholmod_print_common' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        CHOLMOD_print_common ("for UMFPACK", &cm) ;
        ^
/Users/fwein/Downloads/SuiteSparse-5.13.0-cmake.3/UMFPACK/Source/umf_cholmod.c:28:35: note: expanded from macro 'CHOLMOD_print_common'
#define CHOLMOD_print_common      cholmod_print_common
                                  ^
1 error generated.
```
The setting does WITH_PRINT=OFF and resetting has no effect (possibly because it is a cache variable).

I do not set -Werror but use the default settings. These result in

```
Building C object CMakeFiles/umfpack_int.dir/UMFPACK/Source/umf_cholmod.c.o
/Library/Developer/CommandLineTools/usr/bin/cc -DBLAS_UNDERSCORE -DDINT -Dumfpack_EXPORTS -I/Users/fwein/code/SuiteSparse-serguid/AMD/Include -I/Users/fwein/code/SuiteSparse-serguid/UMFPACK/Include -I/Users/fwein/code/SuiteSparse-serguid/UMFPACK/Source -I/Users/fwein/code/SuiteSparse-serguid/build/include/SuiteSparse/UMFPACK -I/Users/fwein/code/SuiteSparse-serguid/build/CHOLMOD/Include -I/Users/fwein/code/SuiteSparse-serguid/CHOLMOD/Cholesky -I/Users/fwein/code/SuiteSparse-serguid/CHOLMOD/Core -I/Users/fwein/code/SuiteSparse-serguid/CHOLMOD/Include -I/Users/fwein/code/SuiteSparse-serguid/CHOLMOD/MatrixOps -I/Users/fwein/code/SuiteSparse-serguid/CHOLMOD/Modify -I/Users/fwein/code/SuiteSparse-serguid/CHOLMOD/Supernodal -I/Users/fwein/code/SuiteSparse-serguid/build/include/SuiteSparse/CHOLMOD -I/Users/fwein/code/SuiteSparse-serguid/SuiteSparse_config -I/Users/fwein/code/SuiteSparse-serguid/build/include/SuiteSparse -I/Users/fwein/code/SuiteSparse-serguid/build/include/SuiteSparse/AMD -O3 -DNDEBUG -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk -mmacosx-version-min=12.3 -fPIC -fvisibility=hidden -MD -MT CMakeFiles/umfpack_int.dir/UMFPACK/Source/umf_cholmod.c.o -MF CMakeFiles/umfpack_int.dir/UMFPACK/Source/umf_cholmod.c.o.d -o CMakeFiles/umfpack_int.dir/UMFPACK/Source/umf_cholmod.c.o -c /Users/fwein/code/SuiteSparse-serguid/UMFPACK/Source/umf_cholmod.c
/Users/fwein/code/SuiteSparse-serguid/UMFPACK/Source/umf_cholmod.c:224:9: error: implicit declaration of function 'cholmod_print_common' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        CHOLMOD_print_common ("for UMFPACK", &cm) ;
        ^
/Users/fwein/code/SuiteSparse-serguid/UMFPACK/Source/umf_cholmod.c:28:35: note: expanded from macro 'CHOLMOD_print_common'
#define CHOLMOD_print_common      cholmod_print_common
                                  ^
1 error generated.
```
I checked this for clang on macOS and icx on Windows. I'll try your patch. Thanks!

--- a/UMFPACK/Source/umf_cholmod.c
+++ b/UMFPACK/Source/umf_cholmod.c
@@ -219,10 +219,12 @@ int UMF_cholmod
     user_info [2] = flops ;
 
     CHOLMOD_free_factor (&L, &cm) ;
-    if (print_level > 0) 
+#ifndef NCHECK
+    if (print_level > 0)
     {
         CHOLMOD_print_common ("for UMFPACK", &cm) ;
     }
+#endif
     CHOLMOD_finish (&cm) ;
     return (TRUE) ;
 #else

